service-registry/src/main/java/io/servicecomb/serviceregistry/cache/InstanceCacheManagerOld.java

```diff
@@ -89,19 +93,32 @@ private InstanceCache create(String appId, String microserviceName, String micro
     return instCache;
   }
 
-  public InstanceCache createInstanceCache(String appId, String microserviceName, String microserviceVersionRule) {
-    List<MicroserviceInstance> instances =
-        serviceRegistry.findServiceInstance(appId, microserviceName, microserviceVersionRule);
-    if (instances == null) {
+  public InstanceCache createInstanceCache(InstanceCache instanceCache) {
+    String appId = instanceCache.getAppId();
+    String microserviceName = instanceCache.getMicroserviceName();
+    String microserviceVersionRule = instanceCache.getMicroserviceVersionRule();
+    MicroserviceInstanceRefresh microerviceInstanceRefresh =
+        serviceRegistry.findServiceInstance(appId, microserviceName, microserviceVersionRule, instanceCache.getRevision());
+    if (microerviceInstanceRefresh == null) {
       return null;
     }
+    if (!microerviceInstanceRefresh.isNeedRefresh()) {
+      return instanceCache;
+    }
+    String revision = microerviceInstanceRefresh.getRevision();
+    String rev = instanceCache.getRevision();
+    if (Integer.valueOf(rev) - Integer.valueOf(revision) >= 0) {
```

Comment:
SC said that rev is a String
is it safe to convert to int?

Reply:
SC said the rev will get bigger and bigger,so told us saved to String will be better

Reply:
why bigger has relationship with String?

Reply:
let's discuss with SC to make sure

Reply:
still use String